### PR TITLE
Rename io

### PR DESCRIPTION
**Pull request by MitraDarja**
#520 

Alle renamings for IO.

sequence_file_output_format_concept -> SequenceFileOutputFormat
alignment_file_output_format_concept -> AlignmentFileOutputFormat
structure_file_output_format_concept -> StructureFileOutputFormat
output_format_concept-> OutputFormat
sequence_file_InputFormat -> SequenceFileInputFormat
structure_file_InputFormat -> StructureFileInputFormat
input_format_concept-> InputFormat
ostream_concept -> Ostream
istream_concept -> Istream
stream_concept -> Stream
type_list_of_structure_file_output_formats_concept -> TypeListOfStructureFileOutputFormats
type_list_of_structure_file_input_formats_concept -> TypeListOfStructureFileInputFormats
type_list_of_sequence_file_output_formats_concept ->TypeListOfSequenceFileOutputFormats
type_list_of_sequence_file_input_formats_concept ->TypeListOfSequenceFileInputFormats
type_list_of_alignment_file_output_formats_concept ->TypeListOfAlignmentFileOutputFormats
sequence_file_format_concept -> SequenceFileFormat
sequence_file_input_traits_concept -> SequenceFileInputTraits
sequence_file_format_out_concept -> SequenceFileFormatOut
parse_condition_concept -> ParseCondition

### CODE REVIEW

**Comment by h-2** [comment]
This looks good, just two small things:

1. need to rebase on master
2. ostream_concept -> O**S**~~s~~tream (same for istream)

File: include/seqan3/io/sequence_file/input.hpp [hunk 1]
```diff
@@ -426,21 +426,21 @@ struct sequence_file_input_default_traits_aa : sequence_file_input_default_trait
...
 template <
-    sequence_file_input_traits_concept                       traits_type_        = sequence_file_input_default_traits_dna,
+    SequenceFileInputTraits                                  traits_type_        = sequence_file_input_default_traits_dna,
     detail::fields_concept                                   selected_field_ids_ = fields<field::SEQ,
                                                                                           field::ID,
                                                                                           field::QUAL>,
-    detail::type_list_of_sequence_file_input_formats_concept valid_formats_      = type_list<sequence_file_format_fasta,
-                                                                                          sequence_file_format_fastq
-                                                                                         /*, ...*/>,
+    detail::TypeListOfSequenceFileInputFormats               valid_formats_      = type_list<sequence_file_format_fasta,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/input.hpp]:
```suggestion
    detail::TypeListOfSequenceFileInputFormats  valid_formats_      = type_list<sequence_file_format_fasta,
```

And also here

File: include/seqan3/io/sequence_file/input.hpp [hunk 2]
```diff
@@ -426,21 +426,21 @@ struct sequence_file_input_default_traits_aa : sequence_file_input_default_trait
...
+ * seqan3::SequenceFileInputTraits.
  *
  * ### Formats
  *
  * TODO give overview of formats, once they are all implemented
  */
 
 template <
-    sequence_file_input_traits_concept                       traits_type_        = sequence_file_input_default_traits_dna,
+    SequenceFileInputTraits                                  traits_type_        = sequence_file_input_default_traits_dna,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/input.hpp]:
```suggestion
    SequenceFileInputTraits                    traits_type_        = sequence_file_input_default_traits_dna,
```
You can change the indentation here

File: include/seqan3/io/sequence_file/input.hpp [hunk 3]
```diff
@@ -946,9 +946,9 @@ sequence_file_input(stream_type & stream,
 namespace std
 {
 //!\brief std::tuple_size overload for column-like access. [metafunction specialisation for seqan3::sequence_file_input]
-template <seqan3::sequence_file_input_traits_concept                       traits_type,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/input.hpp]: Alignment

File: include/seqan3/io/sequence_file/input.hpp [hunk 4]
```diff
@@ -958,9 +958,9 @@ struct tuple_size<seqan3::sequence_file_input<traits_type, selected_field_ids, v
 
 //!\brief std::tuple_element overload for column-like access. [metafunction specialisation for seqan3::sequence_file_input]
 template <size_t                                                        elem_no,
-          seqan3::sequence_file_input_traits_concept                       traits_type,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/input.hpp]: Alignment

File: include/seqan3/io/sequence_file/output.hpp [hunk 5]
```diff
@@ -909,8 +909,8 @@ class sequence_file_output
  * \relates seqan3::sequence_file_output
  * \{
  */
-template <ostream_concept2                     stream_t,
-          sequence_file_output_format_concept file_format,
+template <OStream2                            stream_t,
+          SequenceFileOutputFormat            file_format,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/output.hpp]: Alignment

File: include/seqan3/io/sequence_file/output.hpp [hunk 6]
```diff
@@ -919,8 +919,8 @@ sequence_file_output(stream_t &&,
                             type_list<file_format>,
                             typename std::remove_reference_t<stream_t>::char_type>;
 
-template <ostream_concept2                     stream_t,
-          sequence_file_output_format_concept file_format,
+template <OStream2                             stream_t,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/output.hpp]: Alignment

File: include/seqan3/io/structure_file/InputFormat.hpp [hunk 7]
```diff
@@ -38,7 +38,7 @@ namespace seqan3
  */
 //!\cond
 template<typename t>
-SEQAN3_CONCEPT structure_file_input_format_concept = requires(t & v,
+SEQAN3_CONCEPT StructureFileInputFormat =          requires(t & v,
```
Comment (eseiler) [inline include/seqan3/io/structure_file/InputFormat.hpp]: Remove unnecessary whitespaces

File: include/seqan3/io/structure_file/OutputFormat.hpp [hunk 8]
```diff
@@ -37,7 +37,7 @@ namespace seqan3
  */
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT structure_file_output_format_concept = requires(t & v,
+SEQAN3_CONCEPT StructureFileOutputFormat =          requires(t & v,
```
Comment (eseiler) [inline include/seqan3/io/structure_file/OutputFormat.hpp]: remove whitespaces

File: include/seqan3/io/structure_file/output.hpp [hunk 9]
```diff
@@ -914,8 +914,8 @@ class structure_file_out
  * \relates seqan3::structure_file_out
  * \{
  */
-template <ostream_concept<char>                stream_type,
-          structure_file_output_format_concept file_format,
+template <OStream<char>                        stream_type,
```
Comment (eseiler) [inline include/seqan3/io/structure_file/output.hpp]: alignment

File: include/seqan3/io/sequence_file/InputFormat.hpp [hunk 10]
```diff
@@ -38,7 +38,7 @@ namespace seqan3
  */
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT sequence_file_input_format_concept = requires (t                                     & v,
+SEQAN3_CONCEPT SequenceFileInputFormat = requires       (t                                     & v,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/InputFormat.hpp]: Remove unnecessary whitespaces

File: include/seqan3/io/sequence_file/OutputFormat.hpp [hunk 11]
```diff
@@ -37,7 +37,7 @@ namespace seqan3
  */
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT sequence_file_output_format_concept = requires (t                         & v,
+SEQAN3_CONCEPT SequenceFileOutputFormat = requires       (t                         & v,
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/OutputFormat.hpp]: whitespaces

File: include/seqan3/io/sequence_file/input_format_concept.hpp [hunk 12]
```diff
@@ -38,13 +38,13 @@ namespace seqan3
...
-                                                         std::string                           & id,
-                                                         std::vector<phred42>                  & qual,
-                                                         std::vector<dna5q>                    & seq_qual)
+SEQAN3_CONCEPT SequenceFileInputFormat = requires (t                                     & v,
+                                                   std::ifstream                         & f,
+                                                   sequence_file_input_options<dna5, false> & options,
+                                                   dna5_vector                           & seq,
+                                                   std::string                           & id,
+                                                   std::vector<phred42>                  & qual,
+                                                   std::vector<dna5q>                    & seq_qual)
```
Comment (h-2) [inline include/seqan3/io/sequence_file/input_format_concept.hpp]: If you already touch all of these line, can you align the `&` to be in one column? That was the original plan!

File: include/seqan3/io/sequence_file/output_format_concept.hpp [hunk 13]
```diff
@@ -37,13 +37,13 @@ namespace seqan3
...
-                                                          std::string               & id,
-                                                          std::vector<phred42>   & qual,
-                                                          std::vector<dna5q>        & seq_qual)
+SEQAN3_CONCEPT SequenceFileOutputFormat = requires (t                         & v,
+                                                    std::ofstream             & f,
+                                                    sequence_file_output_options & options,
+                                                    dna5_vector               & seq,
+                                                    std::string               & id,
+                                                    std::vector<phred42>      & qual,
+                                                    std::vector<dna5q>        & seq_qual)
```
Comment (h-2) [inline include/seqan3/io/sequence_file/output_format_concept.hpp]: same here

**Comment by h-2** [comment]
LGTM but there is a conflict that needs to be resolved now!

**Pull request merged**